uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -260,12 +261,19 @@ async def asgi_send(self, message: "ASGISendEvent") -> None:
                 self.accepted_subprotocol = cast(
                     Optional[Subprotocol], message.get("subprotocol")
                 )
-                if "headers" in message:
-                    self.extra_headers.extend(
-                        # ASGI spec requires bytes
-                        # But for compatibility we need to convert it to strings
-                        (name.decode("latin-1"), value.decode("latin-1"))
-                        for name, value in message["headers"]
+                headers = list(message.get("headers", []))
+                _added_names = []
+                for name, value in headers:
+                    if name.lower() in _added_names:
+                        continue
+                    _added_names.append(name.lower())
+                    self.extra_headers.append(
+                        (
+                            # ASGI spec requires bytes
+                            # But for compatibility we need to convert it to strings
+                            name.decode("latin-1"),
+                            value.decode("latin-1"),
+                        )
```

Comment:
You can use the master branch of websockets on this PR, and when they release it we just need to change the version..

Reply:
I have added. Updated the dependency is `setup.py` to point to the master repo of Websockets to run the tests. 

Websockets does not allow `--no-date-header` and adding multiple `Server` headers.